app/Jobs/TranscodeVideo.php

```diff
@@ -112,13 +105,14 @@ public function failed(?Throwable $exception): void
     {
         // All properties have not yet been initialized, because failed jobs use a new instance.
 
-        $tempDerivativesDirectoryPath = $this->getTempVideoDerivativesDirectoryPath();
         $localDisk = Storage::disk('local');
 
-        MediaStorage::VIDEO_DERIVATIVES->getDisk()->deleteDirectory($tempDerivativesDirectoryPath);
+        MediaStorage::VIDEO_DERIVATIVES->getDisk()->deleteDirectory($this->getTempDerivativesDirectoryPath());
         $localDisk->delete($this->getTempMp4Filename());
         $localDisk->delete($this->getTempOriginalFilename());
         $localDisk->deleteDirectory($this->getFfmpegTempDirectory());
+        // This directory stores local temp derivatives in case cloud storage is used.
+        $localDisk->deleteDirectory($this->version->Media->User->name);
```

Comment:
maybe not kill everything specific to the user that might be running, but something specific to this operation

Reply:
I'm now only deleting the temp derivatives folder for the current version.